### ISSUE TITLE
Azure PowerShell Add Option to call Enable-AzureRmAlias before executing script 

### ISSUE THREAD

**Issue by mozts2005** [microsoft/azure-pipelines-tasks#10262]
When trying to migrate to using the new Azure PowerShell v 4 task. We are not able to update all scripts at this time.  We would like to have Azure Powershell task to have the Option to call Enable-AzureRmAlias.

```powershell
Enable-AzureRmAlias -Scope CurrentUser
````

We have tried to enable this option on a private build agent, but at this time the user profile is not loaded when running the Azure PowerShell task.

Our project has some 300 Powershell files so updating them call AzureRmAlias setting is not an option. 

We do plan to convert to Az module but, can't at this time. We do have a few scripts that require Az module and as has been noted in issue #7898 both AzureRm and Az can't be installed on the same agent at a time.  We want to update our agent to have the Az module and have the AzureRm alias turned on so we can update our script as we have time.  

Based on our testing we confident the change can be made. I am just not sure it can be backward compatible.

I would be willing to make a pass at PR  want to make sure this is something the team would want.

**Comment by 20shivangi** [comment]
@mozts2005 We are fixing this issue. Once fixed, we will let you know.

**Comment by 20shivangi** [comment]
We have fixed the issue , now the warning that Az and AzureRM modules are installing side by side will not come. Now, Enable-AzureRmAlias will work successfully. This change may take 3-4 weeks to get deployed.

**Comment by iricigor** [comment]
Please review this closing. @20shivangi 

Side-by-side installation is not what was requested. Notice the sentence "Our project has some 300 Powershell files so updating them call AzureRmAlias setting is not an option." @mozts2005 

P.S. I have the same issue, and I do not want to modify all scripts.

**Comment by 20shivangi** [comment]
@iricigor 
Currently there is no plan to add "Enable-AzureRmAlias" as input in Azure PowerShell V4 task. We recommend users to use V3 task in case the script is not yet updated yet.
You can use V4 tasks for the scripts which is already updated with Az commands and keep using V3 task for old scripts.

In case you are using private build agent (not hosted agent), you can add two agent where one agent will have AzureRM installed and other one will have Az module installed.

In case you want to use same build agent, please install Az module and same AzureRM module in C:\Modules\azurerm_{version} (example:- C:\Modules\azurerm_6.7.0) and set the PowerShell version to the specific version (ex:- 6.7.0). Please refer to the snapshot attached.

![image](https://user-images.githubusercontent.com/45754171/65759006-adfcfd80-e137-11e9-987f-663ac5d2c486.png)

**Comment by 20shivangi** [comment]
Feel free to open the issue, if face any

**Comment by ruijarimba** [comment]
I have the same issue as @mozts2005.

Migrating all scripts (probably hundreds) to use Az module is not an option. Also, adding `Enable-AzureRMAlias` to the top of each script is not a great idea, at least according to me. 

`You can use V4 tasks for the scripts which is already updated with Az commands and keep using V3 task for old scripts.` - this doesn't seem a good idea as well as we don't want to have different versions of the build agents (some with AzureRM module, others with Az module)

If we can't have Az and AzureRM modules installed side-by-side I'd expect, at least, to be able to use Enable-AzureRMAlias in the build agents (in our case, private build agents running as Docker containers). Azure Powershell task will run with the -NoProfile flag, which prevent us to to use the aliases created by that command.


In short, what we need is:
- to replace module AzureRM with module Az AND
- find a way to running old scripts, ideally without adding Enable-AzureRMAlias to every script

Proposed solutions:
- Add a new version of the Azure Powershell task that can run WITHOUT the -NoProfile flag
- Add a new version of the Azure Powershell task that allow us to use Enable-AzureRMAlias
- Add a new version of the Azure Powershell task that allow us to select the scope (LocalMachine, CurrentUser, ...)
